lib/handler/RedirectHandler.js

```diff
@@ -38,6 +38,7 @@ class RedirectHandler {
     this.maxRedirections = maxRedirections
     this.handler = handler
     this.history = []
+    this.redirectionLimitReached = false
```

Comment:
This flag is the one that the implementer should provide to enable the feature. If the caller sets it to `false` it means that doesn't want to have that feature and would rather prefer the old behaviour.

For instance we'll need to document it as we are exposing it, also remember to add it to the TS types of the `RedirectHandler`